Add Mozilla's gaplugin.Linker shim

kzar opened the pull request:
Some websites break if the gaplugin.Linker API isn't present. Our
Google Analytics "surrogate" script didn't shim that API yet, so let's
start doing that now. Instead of reinventing the wheel here, let's
reuse Mozilla's shim[1].

1 - https://searchfox.org/mozilla-central/source/browser/extensions/webcompat/shims/google-analytics-and-tag-manager.js

kdzwinel commented on surrogates/ga.js:
```diff
@@ -45,4 +45,23 @@
+
+    // gaplugin wrapper based on Mozilla's Google analytics shim:
+    // https://searchfox.org/mozilla-central/source/browser/extensions/webcompat/shims/google-analytics-and-tag-manager.js
+    if (!window?.gaplugins?.Linker) {
```
is it safe for us to use fancy `?.` syntax for surrogates?

kzar replied:
I hope so, because we use much fancier syntax in content-scope-scripts. It certainly is OK for the browser extension, but I don't know how to verify it for all of our browsers though. I'll just rewrite it to be safe, we don't benefit from taking a risk.

kzar replied:
Done.

dharb reviewed (changes requested):
Hey @kzar - this change should go in https://github.com/duckduckgo/tracker-surrogates/blob/main/surrogates/analytics.js instead of here, right? It's a little confusing since there are two different Google Analytics surrogates, but the breakage we ran into in https://github.com/duckduckgo/privacy-configuration/pull/1381 was related to the other surrogate.

Mind testing with that site to make sure this resolves the breakage?

kzar commented:
> Hey @kzar - this change should go in https://github.com/duckduckgo/tracker-surrogates/blob/main/surrogates/analytics.js instead of here, right? It's a little confusing since there are two different Google Analytics surrogates, but the breakage we ran into in [duckduckgo/privacy-configuration#1381](https://github.com/duckduckgo/privacy-configuration/pull/1381) was related to the other surrogate.
> 
> Mind testing with that site to make sure this resolves the breakage?

I did already test that this change fixed the issue for the website you linked. I disabled the mitigation, reproduced the issue, then even ran the code through a debugger after a change. So it definitely did work. Perhaps both surrogates are in play on that website though, let me double-check.

kzar commented:
> > Hey @kzar - this change should go in https://github.com/duckduckgo/tracker-surrogates/blob/main/surrogates/analytics.js instead of here, right? It's a little confusing since there are two different Google Analytics surrogates, but the breakage we ran into in [duckduckgo/privacy-configuration#1381](https://github.com/duckduckgo/privacy-configuration/pull/1381) was related to the other surrogate.
> > Mind testing with that site to make sure this resolves the breakage?
> 
> I did already test that this change fixed the issue for the website you linked. I disabled the mitigation, reproduced the issue, confirmed the change fixed the issue, then even ran the code through a debugger after the change. So it definitely did work. Perhaps both surrogates are in play on that website though, let me double-check.

@dharb Sorry, you're completely right there, fixed.

Turned out I had removed the mitigation from the config using the `devtools-panel.html` page, then when I reloaded the extension to test my fix, the changes to the config were lost. So I then saw the website working again, but not because the fix worked. Remind me to just bundle the test config in my test build next time. I'm glad you caught it anyway.

dharb reviewed (approved):
👍 thanks! Looks good now.
